syntax = "proto3";

service XFTServer {
  rpc predict (QueryIds) returns (ResponseIds) {}

  rpc predict_stream (QueryIds) returns (stream ResponseIds) {}
}

message QueryIds {
  repeated int32 Ids = 1;
  int32 batch_size = 2;
  int32 seq_len = 3;
}

message ResponseIds {
  repeated int32 Ids = 1;
  int32 batch_size = 2;
  int32 seq_len = 3;
}